The "//IGNORE" suffix is GNU C/GNU libiconv extension not supported by illumos.

--- cups-2.4.1/cups/transcode.c.orig	2022-01-27 12:11:42.000000000 +0000
+++ cups-2.4.1/cups/transcode.c	2022-08-21 08:43:14.761916178 +0000
@@ -150,7 +150,7 @@
 
     _cupsCharmapFlush();
 
-    snprintf(toset, sizeof(toset), "%s//IGNORE", _cupsEncodingName(encoding));
+    snprintf(toset, sizeof(toset), "%s", _cupsEncodingName(encoding));
 
     map_encoding  = encoding;
     map_from_utf8 = iconv_open(_cupsEncodingName(encoding), "UTF-8");
@@ -279,7 +279,7 @@
 
     _cupsCharmapFlush();
 
-    snprintf(toset, sizeof(toset), "%s//IGNORE", _cupsEncodingName(encoding));
+    snprintf(toset, sizeof(toset), "%s", _cupsEncodingName(encoding));
 
     map_encoding  = encoding;
     map_from_utf8 = iconv_open(_cupsEncodingName(encoding), "UTF-8");
